migrations/V18__Add_Plan_Favorite.sql

```diff
@@ -0,0 +1,19 @@
+CREATE TABLE plan_favorite (
+    id UUID PRIMARY KEY NOT NULL,
+    model_plan_id UUID NOT NULL, --foreign key to model plan
+    user_id EUA_ID NOT NULL,
+    --META DATA
+    created_by EUA_ID NOT NULL,
+    created_dts TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    modified_by EUA_ID,
+    modified_dts TIMESTAMP WITH TIME ZONE
+
+);
+ALTER TABLE plan_favorite
+ADD CONSTRAINT fk_favorite_plan FOREIGN KEY (model_plan_id)
+REFERENCES public.model_plan (id) MATCH SIMPLE
+ON UPDATE NO ACTION
+ON DELETE NO ACTION;
+
+CREATE UNIQUE INDEX idx_plan_favorite_user
+ON plan_favorite(model_plan_id, user_id);
```

Comment:
Postgres will automatically have an index on just `id` right? Since it's a primary key?

Reply:
@ClayBenson94, correct! It will already have a unique index on ID. This index is just to enforce that there is only one plan favorite per user per model. It should also make the query to get by model plan ID and user_id more performant.